common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java

```diff
@@ -15,12 +15,12 @@
 import com.wynntils.utils.mc.type.PoiLocation;
 import com.wynntils.utils.render.Texture;
 import com.wynntils.utils.type.Pair;
-import java.util.LinkedHashSet;
 import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.stream.Stream;
 
 public class UserWaypointMarkerProvider implements MarkerProvider<WaypointPoi> {
-    private final Set<Pair<MarkerInfo, WaypointPoi>> markerInfoSet = new LinkedHashSet<>();
+    private final Set<Pair<MarkerInfo, WaypointPoi>> markerInfoSet = new CopyOnWriteArraySet<>();
```

Comment:
Can you expand a bit about why this was needed? Preferably as a short comment in the code.

Reply:
Well, the marker provider implementations need to be race-proof.

Reply:
https://github.com/Wynntils/Artemis/pull/2125/commits/0488ef8de25077db8aa29a79b49be1990d715caf